src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/ExternalMovementsMigrationIntTest.kt

```diff
@@ -60,6 +72,77 @@ class ExternalMovementsMigrationIntTest(
       externalMovementsNomisApi.verifyGetTemporaryAbsences(offenderNo = "A0001KT")
       externalMovementsNomisApi.verifyGetTemporaryAbsences(offenderNo = "A0002KT")
     }
+
+    @Test
+    fun `will create mappings`() {
+      mappingApi.verify(
+        postRequestedFor(urlEqualTo("/mapping/temporary-absences"))
+          .withRequestBodyJsonPath("prisonerNumber", "A0001KT")
+          .withRequestBodyJsonPath("migrationId", migrationId),
+      )
+      mappingApi.verify(
+        postRequestedFor(urlEqualTo("/mapping/temporary-absences"))
+          .withRequestBodyJsonPath("prisonerNumber", "A0002KT")
+          .withRequestBodyJsonPath("migrationId", migrationId),
+      )
+    }
+
+    @Test
+    fun `will publish telemetry`() {
+      verify(telemetryClient).trackEvent(
+        eq("external-movements-migration-entity-migrated"),
+        check {
+          assertThat(it["offenderNo"]).isEqualTo("A0001KT")
+          assertThat(it["migrationId"]).isEqualTo(migrationId)
+        },
+        isNull(),
+      )
+      verify(telemetryClient).trackEvent(
+        eq("external-movements-migration-entity-migrated"),
+        check {
+          assertThat(it["offenderNo"]).isEqualTo("A0002KT")
+          assertThat(it["migrationId"]).isEqualTo(migrationId)
+        },
+        isNull(),
+      )
+    }
+  }
+
+  @Nested
+  inner class MappingErrorRecovery {
+    @BeforeEach
```

Comment:
Also need to handle mapping already exists and duplicate mapping on create - coming soon